lib/foreman/event_subscribers.rb

```diff
@@ -2,7 +2,14 @@ module Foreman
   module EventSubscribers
     def self.all_observable_events
       # in development descendants will only return classes if they have been loaded
-      @all_observable_events ||= ActiveRecord::Base.descendants.select { |klass| klass <= ::Foreman::ObservableModel }.map(&:event_subscription_hooks).flatten.uniq
+      @all_observable_events ||= begin
+        if Rails.env.development?
+          Rails.logger.debug "Performing eager load to find out all observable classes"
+          Rails.application.eager_load!
+        end
+        ApplicationRecord.descendants.select { |klass| klass <= ::Foreman::ObservableModel }.map(&:event_subscription_hooks) +
+        ApplicationJob.descendants.select { |klass| klass <= ::Foreman::ObservableJob }.map(&:event_subscription_hooks)
```

Comment:
Could you also introduce a method to this class, which would serve as an extension point for plugins wishing to add observable events not covered by `ObservableModel` and `ObservableJob`?

Reply:
I think at some point we should add arbitrary event, something that can be called explicitly and is not either model or job callback. I'd say it's out of scope, I wonder if this should come after refactoring mentioned above as it can shake up things. I created another ticket for that: https://projects.theforeman.org/issues/31647